### PR TITLE
Bugfix: call getToken of refreshTokenGrantType only if refreshToken exists

### PR DESCRIPTION
**Pull request by sgasser**
If refreshToken is not set it not should call the method getToken, because it call the server with an empty refreshToken.

### CODE REVIEW

**Comment by pjcdawkins** [comment]
Hm. The reason for the previous logic was that you can set the 'refresh_token' directly when constructing the RefreshToken grant type. So this means there is a bit of an API change

**Comment by sgasser** [comment]
But your example is not working, because there is no refreshToken and it works only with an refresh token...

## Example
```php
use GuzzleHttp\Client;
use CommerceGuys\Guzzle\Oauth2\GrantType\RefreshToken;
use CommerceGuys\Guzzle\Oauth2\GrantType\PasswordCredentials;
use CommerceGuys\Guzzle\Oauth2\Oauth2Subscriber;

$base_url = 'https://example.com';

$oauth2Client = new Client(['base_url' => $base_url]);

$config = [
    'username' => 'test@example.com',
    'password' => 'test password',
    'client_id' => 'test-client',
    'scope' => 'administration',
];

$token = new PasswordCredentials($oauth2Client, $config);
$refreshToken = new RefreshToken($oauth2Client, $config);

$oauth2 = new Oauth2Subscriber($token, $refreshToken);

$client = new Client([
    'defaults' => [
        'auth' => 'oauth2',
        'subscribers' => [$oauth2],
    ],
]);

$response = $client->get('https://example.com/api/user/me');

print_r($response->json());

// Use $oauth2->getAccessToken(); and $oauth2->getRefreshToken() to get tokens
// that can be persisted for subsequent requests.

**Comment by sgasser** [comment]
I think we can fix the bug with this solutions:
1. Oauth2Subscriber.php: if in onBefore method ```$request->getConfig()->get('refresh_token')``` is set we set ```$this->refreshToken = $request->getConfig()->get('refresh_token')```
2. We override getToken method in RefreshToken.php and call it only if refresh_token is set.

**Comment by sgasser** [comment]
Any news?

**Comment by pjcdawkins** [comment]
This was what I was thinking of, with an explicit method to check that the refresh token is available:
https://github.com/commerceguys/guzzle-oauth2-plugin/compare/master...pjcdawkins:check-refresh-token-set

So the subscriber would only try the refresh token grant type if it has a non-empty refresh token set. This seems a bit wordy, but it would do the job.

**Comment by sgasser** [comment]
Cool, it works!

After merge https://github.com/commerceguys/guzzle-oauth2-plugin/compare/master...pjcdawkins:check-refresh-token-set we can close this pr.

**Comment by pjcdawkins** [comment]
Merged in https://github.com/commerceguys/guzzle-oauth2-plugin/commit/663ba6c893a4b097205257089d2a18b6918e0700

So that can be in a v2.0.1 release, which I'll make at the end of the day after a bit of testing. Thanks!
